code/modules/multiz/turf.dm

```diff
@@ -1,90 +1,48 @@
-/turf/var/list/zstructures	//A list of structures that may allow or interfere with ztransitions
-
-/atom/proc/CanZPass(atom/A, direction)
-	return TRUE
-
-/turf/CanZPass(atom/A, direction)
+/turf/proc/CanZPass(atom/A, direction)
 	if(z == A.z) //moving FROM this turf
-		.= direction == UP //can't go below
+		return direction == UP //can't go below
 	else
 		if(direction == UP) //on a turf below, trying to enter
-			.=FALSE
+			return FALSE
 		if(direction == DOWN) //on a turf above, trying to enter
-			.= !density
-
-	if (LAZYLEN(zstructures))
-		var/highest_priority = 0
-		for (var/atom/B in zstructures)
-			if (zstructures[B] > highest_priority)
-				var/result=B.CanZPass(A, direction)
-				if (result != ZTRANSITION_MAYBE)
-					highest_priority = zstructures[B]
-					.=result
+			return !density
 
 /turf/simulated/open/CanZPass(atom/A, direction)
-	.=TRUE
-	if (LAZYLEN(zstructures))
```

Comment:
this file has the zstructure optimisation I was talking about. this needs to be preserved in some form

Reply:
> this file has the zstructure optimisation I was talking about. this needs to be preserved in some form

ready